File: SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7121Form.ts
import { StringEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface Pfk7121Form {
    ColorName: StringEditor;
    ColorNameS1: StringEditor;
    ColorNameS2: StringEditor;
    ColorNameSimple: StringEditor;
    CustomerCode: StringEditor;
    SeColorBase: StringEditor;
    CdColorBase: StringEditor;
    SeColorCategory: StringEditor;
    CdColorCategory: StringEditor;
    ColorPosition: StringEditor;
    CheckBase: StringEditor;
    CheckUse: BooleanEditor;
    DateInsert: StringEditor;
    DateUpdate: StringEditor;
    InchargeInsert: StringEditor;
    InchargeUpdate: StringEditor;
    Remark: StringEditor;
    SeSite: StringEditor;
    CdSite: StringEditor;
}

export class Pfk7121Form extends PrefixedContext {
    static readonly formKey = 'Default.Pfk7121';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!Pfk7121Form.init)  {
            Pfk7121Form.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(Pfk7121Form, [
                'ColorName', w0,
                'ColorNameS1', w0,
                'ColorNameS2', w0,
                'ColorNameSimple', w0,
                'CustomerCode', w0,
                'SeColorBase', w0,
                'CdColorBase', w0,
                'SeColorCategory', w0,
                'CdColorCategory', w0,
                'ColorPosition', w0,
                'CheckBase', w0,
                'CheckUse', w1,
                'DateInsert', w0,
                'DateUpdate', w0,
                'InchargeInsert', w0,
                'InchargeUpdate', w0,
                'Remark', w0,
                'SeSite', w0,
                'CdSite', w0
            ]);
        }
    }
}

File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/Default/Pfk7121/Pfk7121Form.cs
using Serenity.ComponentModel;

namespace SerenePSM.Default.Forms;

[FormScript("Default.Pfk7121")]
[BasedOnRow(typeof(Pfk7121Row), CheckNames = true)]
public class Pfk7121Form
{
    public string ColorName { get; set; }
    public string ColorNameS1 { get; set; }
    public string ColorNameS2 { get; set; }
    public string ColorNameSimple { get; set; }
    public string CustomerCode { get; set; }
    public string SeColorBase { get; set; }
    public string CdColorBase { get; set; }
    public string SeColorCategory { get; set; }
    public string CdColorCategory { get; set; }
    public string ColorPosition { get; set; }
    public string CheckBase { get; set; }
    public string CheckUse { get; set; }
    public string DateInsert { get; set; }
    public string DateUpdate { get; set; }
    public string InchargeInsert { get; set; }
    public string InchargeUpdate { get; set; }
    public string Remark { get; set; }
    public string SeSite { get; set; }
    public string CdSite { get; set; }
}